### PR TITLE
Remove use of non-public C functions in ctypes_patch

### LINKED ISSUE #113: How to run rubicon-objc on linux
I got the following error

> AttributeError: python: undefined symbol: PyType_stgdict

when I tried to import `rubicon.objc` on Ubuntu. Is there a way to use this library on Linux? If not currently, what do I need to do to make it work?

### PR DESCRIPTION
**Pull request by dgelessus**
This should fix #113.

Also includes a small update to the tox.ini to remove the dependency on the development version of pyflakes.

### CODE REVIEW

**Comment by Jie-Cheng** [comment]
@dgelessus thanks for the fix! I can't wait to use it so I pulled your code. I'm not familiar with python packages, can you guide me to build and use `rubicon` from source? I understand your current `Makefile` is written for mac, I can test it on a mac first.

File: tox.ini [hunk 1]
```diff
@@ -25,10 +25,11 @@ setenv =
 
 [testenv:flake8]
 basepython = python3.6
-# FIXME Remove the pyflakes git dependency once a new version of pyflakes (> 1.6.0) is released.
-# Currently we need to use the development version of pyflakes in order to use __class__, because pyflakes <= 1.6.0
-# doesn't understand __class__. This has been fixed in the development version, but not in any release yet.
+# FIXME Remove the manual pyflakes dependency once flake8 depends on pyflakes >= 2.0.0.
+# Currently we need to manually update pyflakes in order to use __class__, because pyflakes <= 1.6.0
+# doesn't understand __class__. This has been fixed in pyflakes 2.0.0, but flake8's requirements don't allow that
+# version of pyflakes yet.
```
[the remark this answers is not in the document]
Comment (dgelessus) [reply tox.ini]: Yes, [flake8's current setup.py](https://gitlab.com/pycqa/flake8/blob/33e7eae53ad513952da777cabd3654219efe0eb8/setup.py#L24) only allows `pyflakes >= 1.5.0, < 1.7.0`. The reason for the upper version limit is explained [in the flake8 FAQ](https://flake8.readthedocs.io/en/latest/faq.html#why-does-flake8-use-ranges-for-its-dependencies).

[the remark this answers is not in the document]
Comment (freakboy3742) [reply tox.ini]: Oh - right, now I get it. This is about forcing the pyflakes version because of flake8's stated requirement.

**Comment by freakboy3742** [comment]
@dgelessus FYI - I tried out this patch over the weekend and had no success; however, I wasn't able to get my other "working" example (which commented out the line from the ctypes_patch that cause problems) to work, either...

The issue is that this problem only manifests on AdHoc and App Store builds, those take 5-10 minutes *each time* to prepare, and you completely lose all stdin/stdout debugging assistance when that happens (because, in Apple's wisdom, they've discard all stdin/stdout in favour of their infernal logging framework). I'll have to keep digging...

**Comment by freakboy3742** [comment]
@dgelessus In the past I've hacked together something messy to do exactly that.  Historically, I've done it at the C level so that I can catch errors that come up as part of Rubicon instantiation; although as long as we can install the handler early enough, I suppose a Python implementation would be sufficient.

**Pull request merged**